pkg/model/iam/tests/iam_builder_master_strict.json

```diff
@@ -0,0 +1,165 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeInstances",
+        "ec2:DescribeRouteTables",
+        "ec2:DescribeSecurityGroups",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeVolumes"
+      ],
+      "Resource": [
+        "*"
```

Comment:
this should not be *, and we limit by subnet that is tagged?  Subnet has to be tagged, vpc technically does not.  Lots can live inside a VPC, so if we can target by subnet ...  If we have to do VPC, it is better than nothing.

Reply:
@lwoggardner I brought up an issue a few weeks ago to the team regarding ```ec2:Describe*``` from an attacker's perspective.  Should a pod be compromised on a cluster not blocking EC2 Metadata access, that pod will be able to perform "aws ec2 describe-*" actions including ```aws ec2 describe-instances``` to get all instance-ids in the entire account followed by ```aws ec2 describe-instance-attribute --instance-id i-0123456789 --attribute userData```.  This gives the node the ability to find and then read the ```user-data``` script of any instance in the account (which often contain sensitive items for bootstrapping).  Limiting the ```Describe``` permissions to only what is needed can protect the security of other instances in the account that operators/admins won't know they exposed with this IAM policy.  Cheers 👍